doc/tut1.rst

```diff
@@ -874,6 +876,30 @@ The example also shows that a proc's body can consist of a single expression
 whose value is then returned implicitly.
 
 
+Funcs and methods
+-----------------
+
+As mentioned in the introduction, Nim differentiates between procedures,
+functions, and methods, defined by the `proc`, `func`, and `method` keywords
+respectively. In some ways, Nim is a bit more pedantic in its definitions than
+other languages.
+
+Functions are closer to the concept of a pure mathematical
+function, which might be familiar to you if you've ever done functional
+programming. Essentially they can't access global state (except `const`) and can't
+produce side-effects (they are tagged with `{.noSideEffects.}`). They can still
+change their mutable arguments, which are those marked as `var` and any `ref`
+objects.
+
+Unlike procedures, methods are dynamically dispatched. This sounds a bit
+complicated, but is a concept closely related to inheritance and object oriented
+programming. If you overload a procedure (two procedures with the same name but
+different types are said to be overloaded), the procedure to use is determined
```

Comment:
```suggestion
of different types or with different sets of arguments are said to be overloaded), the procedure to use is determined
```
Being pedantic, a proc with different args is of a different type by definition, but this wording captures both the args and the result type unambiguously.